components/AccessControl.php

```diff
@@ -22,7 +27,8 @@ public function beforeAction($action)
     {
         $actionId = $action->getUniqueId();
         $user = Yii::$app->getUser();
-        if ($user->can('/' . $actionId)) {
+        $params = isset($this->params[$action->id]) ? $this->params[$action->id] : [];
+        if ($user->can('/' . $actionId, $params)) {
```

Comment:
first <code>can</code> checks action (<code>module/controller/action</code>), the second <code>can</code> checks controller (<code>module/controller/\*</code>) and module (<code>module/\*</code>). It's quite enaugh to pass params in first <code>can</code> only, because biz rules generally are not used for controller and module routes.